jax/_src/numpy/polynomial.py

```diff
@@ -242,9 +242,66 @@ def polyadd(a1: Array, a2: Array) -> Array:
     return a2.at[-a1.shape[0]:].add(a1)
 
 
-@implements(np.polyint)
 @partial(jit, static_argnames=('m',))
 def polyint(p: Array, m: int = 1, k: int | None = None) -> Array:
+  r"""Returns the coefficients of the integration of specified order of a polynomial.
+
+  JAX implementation of :func:`numpy.polyint`.
+
+  Args:
+    p: An array of polynomial coefficients.
+    m: Order of integration. Default is 1.
+    k: Sequence or array of ``m`` integration constant (s).
+
+  Returns:
+    An array of coefficients of integrated polynomial.
+
+  Note:
+    Since the size of the output of :func:`jax.numpy.polyint` depends on the
+    value of ``m``, the function semantics are not typically compatible with
+    :func:`~jax.jit()` and other JAX transformations.
```

Comment:
I think this note is misleading. `polyint` is JIT-compiled by default! It's true that `m` must be static, but we can just mention that in its description.

We generally reserve this kind of note for functions whose default semantics are incompatible with JIT (e.g. `jnp.unique` cannot be JIT-compiled without setting a JAX-only argument to specify a static size).

Reply:
Removed.